### PR TITLE
fix(basename): ignore trailing seperator

### LINKED ISSUE #179: `pathe.basename` behaves differently w/ `path.basename` when handling trailing separator
### Environment

Node.js 18 & 22

### Reproduction

```ts
const path = require("path");
const pathe = require("pathe");

console.log({
  path: path.basename("sbin/a/"),
  pathe: pathe.basename("sbin/a/"),
});

// { path: 'a', pathe: '' }

console.log({
  path: path.basename("sbin/a///"),
  pathe: pathe.basename("sbin/a///"),
});

// { path: 'a', pathe: '' }
```

https://replit.com/@isukkaw/ColossalGloriousWamp

### Describe the bug

`pathe.basename("sbin/a/")` should return `a` instead of an empty string.

### Additional context

Per Node.js `basename` documents: https://nodejs.org/api/path.html#pathbasenamepath-suffix

> Trailing [directory separators](https://nodejs.org/api/path.html#pathsep) are ignored.

### Logs

_No response_

### PR DESCRIPTION
**Pull request by SukkaW**
Fixes #179.

### CODE REVIEW

File: src/path.ts [hunk 1]
```diff
@@ -253,7 +253,10 @@ export const format: typeof path.format = function (p) {
 
 // basename
 export const basename: typeof path.basename = function (p, extension) {
-  const lastSegment = normalizeWindowsPath(p).split("/").pop();
+  const lastSegment = normalizeWindowsPath(p)
+    .replace(/\/+$/, "")
+    .split("/")
+    .pop();
```
Comment (pi0) [inline src/path.ts:259]: Perf: I think we can use two lines `const segments = []` and another check to see if last part is empty, fallback to len-1 this way we avoid regex

Comment (SukkaW) [reply src/path.ts:259]: > Perf: I think we can use two lines `const segments = []` and another check to see if last part is empty, fallback to len-1 this way we avoid regex

We need to ignore all trailing sep if there are many. See my test cases (`/foo/bar////`). If we loop and pop, it will be slower than regex.

Comment (pi0) [reply src/path.ts:259]: I think most of the fast and normal conditions is that there is no trailing slash or just one. We don't need to pop, we can just have a reverse for loop to find the first index that has non-empty one.

Feel free to do a benchmark. What matters to me is that we don't add a perf overhead for normal conditions by adding regex replace (since this is a low level utility, it could be used hundreds or thousands of times in high level tools)

Comment (SukkaW) [reply src/path.ts:259]: @pi0 Benchmark: https://replit.com/@isukkaw/SandyPossibleLanguage

TL; DR:

- `endsWith` before regex replace is fastest when the input doesn't end with sep. But it is slowest if there are any
- reverse loop and pop is fastest if there are seps at the end of the input. It is the second fastest if the input doesn't end with one (but it is still 10x slower than `endsWith` before regex replace)

Comment (pi0) [reply src/path.ts:259]: Thanks!! See this updated repl. I have  made few changes, removed other variants and add tests/adjustment that all implementations return basename for us. 

https://replit.com/@pi0/httpsgithubcomunjspathepull180

How do you think?

Comment (SukkaW) [reply src/path.ts]: @pi0 Just a friendly ping~

**Review by pi0 — APPROVED** [review]
❤️
